scalding-core/src/main/java/com/twitter/scalding/cascading_interop/FlowListenerPromise.java

```diff
@@ -67,6 +69,17 @@ public boolean onThrowable(Flow f, Throwable t) {
         return true;
       }
     });
+    flow.addStepListener(new FlowStepListener() {
+      public void onStepStarting(FlowStep flowStep) { } // ignore
+      public void onStepRunning(FlowStep flowStep) { } // ignore
+      public void onStepCompleted(FlowStep flowStep) { } // ignore
+      public void onStepStopping(FlowStep f) { result.tryFailure(new FlowStopException("Flow step was stopped")); }
+      public boolean onStepThrowable(FlowStep f, Throwable t) {
+        result.failure(t);
```

Comment:
Since there can be more than one failure don’t we need tryFailure or something? I think promise errors if two calls are made to fill it unless the try variants are used.

Reply:
ah, you're right. fixed.